### DISCUSSION TITLE
External chat manager?

### DISCUSSION THREAD

**Discussion by skandaI** [rodit/SnapMod#110] [Ideas & Feature Suggestions]
Is it planned to add something like a chat manager? Or some other way to make it possible to save chats (text) permanently regardless of what our chat partner sets?

**Comment by rodit** [comment]
Preventing Snapchat from deleting read messages from your device is something I want to implement at some point, yes (regardless of when messages are configured to be deleted in the chat). I doubt I'd bother with external saving/chat managing though.

**Comment by Magna-Graecia** [comment]
There are already 2 options for this in SnapMod settings: "Hide Chat Saves/Unsaves" and "Don't Release Messages", but unfortunately neither of them work for me at the moment, and I'm pretty sure it's the same for everyone from what I've read so far. But the point of mentioning them is that they show that rodit is clearly interested in implementing this feature. It's just a matter of getting them to work that takes some figuring out.

**Comment by RevealedSoulEven** [comment]
I've requested feature to export Chat. And it's released. So better use it

**Reply by skandaI** [reply]
This solution has the disadvantage that if the chat is set to clear itself after 24 hours, I would have to manually save the conversation to a file every day so that nothing disappears for me.

**Reply by crxed9** [reply]
Yeah the solution would be better if it auto-saves directly to a file outside of SC.

**Reply by RevealedSoulEven** [reply]
Okay! then request a feature to save all incoming/outgoing messages in a chat at the instance it is incoming or outgoing and gets overridden to the 
.txt file. The txt file should be located locally along with the name of the user. @rodit what about it?

**Reply by RevealedSoulEven** [reply]
Even it would be better for chats set to deleted after viewing too

**Reply by rodit** [reply]
I'll look into it

**Reply by RevealedSoulEven** [reply]
Yeaa sure! Please add enhancement tag on it then
